src/System.IO.Pipes/tests/NamedPipeTests/NamedPipeTest.CurrentUserOnly.netcoreapp.cs

```diff
@@ -41,6 +41,20 @@ public static void CreateServer_ConnectClient()
             }
         }
 
+        [Fact]
+        [PlatformSpecific(TestPlatforms.AnyUnix)] // On Unix domain socket should have different location in this case.
+        public static void CreateServerNotCurrentUserOnly_ClientCurrentUserOnly_ThrowsTimeout_OnUnix()
+        {
+            var name = GetUniquePipeName();
+            using (var server = new NamedPipeServerStream(name, PipeDirection.InOut, 1, PipeTransmissionMode.Byte))
+            {
+                using (var client = new NamedPipeClientStream(".", name, PipeDirection.InOut, PipeOptions.CurrentUserOnly))
+                { 
+                    Assert.Throws<TimeoutException>(() => client.Connect(1));
```

Comment:
Is `TimeoutException` really the right type to be throwing here?  It seems like a more helpful exception type could be thrown.

Reply:
Currently the client is not going to find it so that is the "natural" exception, the alternative is to go with `getpeerid` and do not use the directory, see https://github.com/dotnet/corefx/pull/26395#discussion_r166484692 for some comments about it.